Products/ZenUI3/browser/resources/js/zenoss/EventPanels.js

```diff
@@ -1498,6 +1498,11 @@
                     this.applyState(state);
                     this.filterRow.storeSearch();
                 } catch(e) { }
+            // for cases with not encoded parameters
+            } else if (qs) {
+                state = {"filters": Ext.urlDecode(qs)};
```

Comment:
`Ext.urlDecode(qs)` is called here and also just above. it looks like this code could be made more readable by first decoding and saving the result, then checking for existence of `state` on the decoded object.